feat: Introduce Dialog as successor for Modal - Fiori 3 refactor

salarenko opened the pull request:
## Related Issue
Closes SAP/fundamental-styles#512

## Description
Changes introduce:
- new component Dialog replacing Modal
- Dialog component matching Fiori3 standard
- update of Fundamental Styles documentation
- update of developers playground examples
- removed `fd-overlay` support for Modal

Dialog component is composed out of:
- Bar representing dialogs:
    - header
    - subheader
    - footer
- Busy Indicator
    - loader

## Screenshots
![image](https://user-images.githubusercontent.com/17496353/74422647-382bd100-4e4f-11ea-8a4b-6256d8ccc4bf.png)

![image](https://user-images.githubusercontent.com/17496353/74423101-fbaca500-4e4f-11ea-8824-054702af2529.png)

![image](https://user-images.githubusercontent.com/17496353/74423910-54307200-4e51-11ea-9309-0aaaaa41feed.png)

![image](https://user-images.githubusercontent.com/17496353/74424022-84781080-4e51-11ea-97fb-c60264304766.png)

bcullman commented:
not sure if this is in scope, but the box shadow on the modal header *seems* wrong here.

mikerodonnell89 commented:
> not sure if this is in scope, but the box shadow on the modal header _seems_ wrong here.

The box shadow from the specs: 
<img width="294" alt="Screen Shot 2020-01-17 at 1 20 20 PM" src="https://user-images.githubusercontent.com/2471874/72636129-68ee1880-392c-11ea-8f62-40cdeef49785.png">

From this PR: 
<img width="819" alt="Screen Shot 2020-01-17 at 1 24 17 PM" src="https://user-images.githubusercontent.com/2471874/72636294-be2a2a00-392c-11ea-90cf-0984c27ec2ef.png">


I agree that it does seem too pronounced in the PR.  The spec sheet does say to use `--sapContent_Shadow3` which this PR is doing... however I think the spec sheet might've listed the wrong box shadow value

bcullman commented:
Does UI5 web components have it wrong here:

https://sap.github.io/ui5-webcomponents/playground/components/Dialog/

(they also show a centered title)

InnaAtanasova commented on src/modal.scss:
```diff
@@ -4,111 +4,190 @@
+
+    .#{$block}__title {
+      @include fd-reset();
+      @include fd-type("1");
```
Should get rid of `fd-type`, it's a hard-coded value and won't be affected by the theme. You can use css variable from sap-theming

InnaAtanasova commented on src/modal.scss:
```diff
@@ -4,111 +4,190 @@
-      left: fd-space(2);
-      right: auto;
+      @include fd-rtl() {
+        margin-right: fd-space(reg);
```
needs to be `rem` values

InnaAtanasova commented:
<img width="1426" alt="Screen Shot 2020-01-27 at 4 13 32 PM" src="https://user-images.githubusercontent.com/39598672/73214370-082ac100-4120-11ea-8726-f11494f7c63a.png">
I think something is wrong with the width of the modal. The min-width is 20rem which is 320px. This means that the example should look like this:
<img width="1427" alt="Screen Shot 2020-01-27 at 4 15 58 PM" src="https://user-images.githubusercontent.com/39598672/73214495-53dd6a80-4120-11ea-93eb-a0c97101ee17.png">

droshev commented:
> approved although rest of comments have to be fixed

if there

stefanoScalzo commented:
@droshev Shouldn't we add another class with the dialog name?

stefanoScalzo commented:
I think that the dynamic paddings should be shown depending on the screen size since the screen sizes are hardcoded. The information is written properly if they are curious. but by including them for the same screen size it is confusing

stefanoScalzo commented:
![Screen Shot 2020-02-12 at 6 35 48 PM](https://user-images.githubusercontent.com/50607147/74387430-82b53b00-4dc6-11ea-9a84-b69c1e495f24.png)
Handle looks broken

stefanoScalzo commented:
is the draggable dialog in fiori?

stefanoScalzo commented:
Are we adding the message dialog as well since we took care of busy dialog?

stefanoScalzo commented:
![Screen Shot 2020-02-12 at 6 53 18 PM](https://user-images.githubusercontent.com/50607147/74388328-2273c880-4dc9-11ea-90e9-bc0715bc6e26.png)
![Screen Shot 2020-02-12 at 6 53 38 PM](https://user-images.githubusercontent.com/50607147/74388329-2273c880-4dc9-11ea-9aad-57b9ec7da135.png)
![Screen Shot 2020-02-12 at 6 54 00 PM](https://user-images.githubusercontent.com/50607147/74388330-2273c880-4dc9-11ea-81fc-8e489ba9b5c1.png)
fix overflows

stefanoScalzo commented:
Shouldn't there be a line/shadow on top of the subheader?

stefanoScalzo commented:
Need to add footer with begin and end specs in documentation

salarenko commented:
@mikerodonnell89 Dialog buttons now also use `.fd-button--emphasized`. To be honest in case of buttons I don't see much difference between docs screenshots and current implementation.

@stefanoScalzo please don't review PRs in [WIP] and use review mode for comments.
1. Modal has been renamed into Dialog
2. In terms of responsive padding I followed [Dialog with responsive padding](https://sapui5.hana.ondemand.com/#/entity/sap.m.Dialog/sample/sap.m.sample.Dialog). Here padding is applied depending on Dialog size. I think this is more logical solution than applying paddings based on view-port size, but it is to be discussed on daily meeting.
3. Has been fixed
4. It doesn't exist in style specs, but again I was suggested by [Dialog (Draggable)](https://sapui5.hana.ondemand.com/#/entity/sap.m.Dialog/sample/sap.m.sample.Dialog)
5. From what I've consulted with other developers, Message Dialog is going to be introduced as separate component (Message Box)
6. I've seen that issue but I dont know if it shouldn't be also fixed on Bar component side. To be discussed on daily meeting.
7. This is Bar component issue
8. Fixed

rengare commented on src/dialog.scss:
```diff
@@ -0,0 +1,170 @@
+        position: absolute;
+        cursor: nwse-resize;
+        right: 0;
+        bottom: -1px;
```
the specification says that bottom should be 0px, did you set bottom to -1 because of incorrect visual representation?

InnaAtanasova commented on src/dialog.scss:
```diff
@@ -0,0 +1,170 @@
+  &__footer {
+    .#{$block}__decisive-button {
+      min-width: $fd-dialog-footer-button-min-width;
+    }
```
Since it's a requirement for all buttons inside Popover and Dialogs maybe it's better to add this requirement in the Bar component. And to "force" this rule we can target directly button elements. Because if the user omits this class the button won't be forced to apply the 4rem min width. Just a suggestion :)

salarenko replied:
The thing is this should be applied only to "Begin" and "End" buttons. Docs are written as the suggest there can be other buttons inside footer which shouldn't be modified.

![image](https://user-images.githubusercontent.com/17496353/74824000-e7f9b680-5307-11ea-9ec9-aa0c287e870e.png)

InnaAtanasova replied:
We can always target the first and the last child. My only concern is that the user can easily omit this class and to be honest it's easy to avoid this class by targeting the first and the last child of the footer/floating footer. Up to you to decide :)

salarenko commented:
Because of pipeline problems, this feature has been moved from fork branch to origin branch `feat/dialog-fiori3_2` and merged with #687
